pkg/ottl/functions.go

```diff
@@ -103,13 +103,30 @@ func (p *Parser[K]) parsePathContext(path *path) (string, []field, error) {
 	}
 
 	if hasPathContextNames {
+		if ok, contextName := p.isPathToContextRootData(path); ok {
+			return contextName, []field{{}}, nil
+		}
 		originalText := buildOriginalText(path)
 		return "", nil, fmt.Errorf(`missing context name for path "%s", possibly valid options are: %s`, originalText, p.buildPathContextNamesText(originalText))
 	}
 
 	return "", path.Fields, nil
 }
 
+// When a path has no dots separators (e.g.: resource), the grammar extracts it into the
+// path.Fields slice, letting the path.Context empty. This function verifies if a given
+// path is accessing any configured (ottl.WithPathContextNames) object root, returning
+// true and the resolved context name.
+func (p *Parser[K]) isPathToContextRootData(pat *path) (bool, string) {
```

Comment:
If we got rid of support for `set(attributes["body"], resource)` and similar paths separately by removing path definitions like https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/a70fa09f8e31f6847cc24f0e8d16b1e78a69c1b6/pkg/ottl/contexts/internal/resource.go#L21, could we get rid of this logic?

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/36872

Reply:
That's nice, removing support to it simplifies the code a lot, thanks! I'll remove this logic.